src/Microsoft.Health.Dicom.SchemaManager.UnitTests/DicomSchemaClientTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.SqlServer.Features.Schema;
+using Microsoft.Health.SqlServer.Features.Schema.Manager;
+using Microsoft.Health.SqlServer.Features.Schema.Manager.Model;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.SchemaManager.UnitTests;
+
+public class DicomSchemaClientTests
+{
+    private readonly IScriptProvider _scriptProvider = Substitute.For<IScriptProvider>();
+    private readonly ISchemaDataStore _schemaDataStore = Substitute.For<ISchemaDataStore>();
+    private readonly ISchemaManagerDataStore _schemaManagerDataStore = Substitute.For<ISchemaManagerDataStore>();
+
+    [Fact]
+    public async void GivenCurrentVersionAboveOne_GetAvailableVersions_ShouldReturnCorrectVersionsAsync()
+    {
+        //Arrange
+        int currentVersion = 5;
+        _schemaManagerDataStore.GetCurrentSchemaVersionAsync(default).Returns(currentVersion);
+
+        var dicomSchemaClient = new DicomSchemaClient(_scriptProvider, _schemaDataStore, _schemaManagerDataStore);
+
+        //Act
+        List<AvailableVersion>? actualVersions = await dicomSchemaClient.GetAvailabilityAsync();
+
+        var expectedVersions = new List<AvailableVersion>();
+        int numberOfAvailableVersions = SchemaVersionConstants.Max - currentVersion + 1;
+        foreach (int version in Enumerable.Range(currentVersion, numberOfAvailableVersions))
+        {
+            expectedVersions.Add(new AvailableVersion(version, string.Empty, string.Empty));
+        }
```

Comment:
```suggestion
        var expectedVersions = Enumerable
            .Range(currentVersion, numberOfAvailableVersions)
            .Select(version => new AvailableVersion(version, $"{version}.sql", $"{version}.diff.sql"))
            .ToList();
```

Reply:
This is much better than what I wrote, thanks! Added in latest commit.